source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/TimeAndDateFucntionTest.cpp

```diff
@@ -0,0 +1,207 @@
+#include <time.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{		
+	TEST_CLASS(UnitTest1)
+	{
+	public:
+		TEST_METHOD(ISO8601TransformTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+            blck->SetText("{{DATE(2017-02-13T20:46:30Z, Short)}}");
+			Assert::AreEqual<string>("02/13/2017",  blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestLong)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Long)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("Monday, February 13, 2017", blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestDefault)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("02/13/2017", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{a{DATE(2017-02-13T20:46:30Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{a{DATE(2017-02-13T20:46:30Z, Short)}}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest2)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Short)}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{{DATE(2017-02-13T20:46:30Z, Short)}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest3)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017a02-13T20:46:30Z, Short)}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{{DATE(2017a02-13T20:46:30Z, Short)}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(TransformToDateTest1)
+		{
+			TextBlock* blck = new TextBlock();
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("02/13/2017", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(TransformToDateTest2)
+		{
+			TextBlock* blck = new TextBlock();
+			string testString = "{{TIME(2017-10-27T18:19:09Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("11:19 AM", blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(TransformToDateTest3)
+		{
+			TextBlock* blck = new TextBlock();
+			string testString = "{{DATE(2017-10-28T01:47:00Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("10/27/2017", blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(TransformToTimeTest)
+		{
+			TextBlock* blck = new TextBlock();
+			string testString = "{{Time(2017-10-28T02:17:00Z, Short)}}";
```

Comment:
>Short [](start = 53, length = 5)

I was under the impression that time does not have a long/short version. The docs just say it's TIME

Reply:
Let's also make sure that it's supposed to be case insensitive.

---
In reply to: [148399685](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148399685) [](ancestors = 148399685)

Reply:
thanks for letting me know! yes, it made no sense to me, so it returns the same result regardless

---
In reply to: [148399685](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148399685) [](ancestors = 148399685)

Reply:
parsing is not case sensitive

---
In reply to: [148399905](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148399905) [](ancestors = 148399905,148399685)

Reply:
The spec for these features is here:
[https://docs.microsoft.com/en-us/adaptive-cards/create/textfeatures](https://docs.microsoft.com/en-us/adaptive-cards/create/textfeatures)


---
In reply to: [148400013](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148400013) [](ancestors = 148400013,148399685)

Reply:
I mean, are we supposed to actually parse things if the function is in lower case? The docs say the functions are in upper case.

---
In reply to: [148400146](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148400146) [](ancestors = 148400146,148399905,148399685)

Reply:
o.k. sounds good to me about case. I think it should be case sensitive. for time, spec doesn't have that, so we can just ignore it?

---
In reply to: [148400224](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148400224) [](ancestors = 148400224,148400013,148399685)

Reply:
I would check how it works in the html visualizer.

---
In reply to: [148614497](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148614497) [](ancestors = 148614497,148400224,148400013,148399685)

Reply:
right, that's good idea. let me do that.

---
In reply to: [148615263](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148615263) [](ancestors = 148615263,148614497,148400224,148400013,148399685)